src/xrpld/app/tx/detail/Payment.cpp

```diff
@@ -357,6 +362,17 @@ Payment::preclaim(PreclaimContext const& ctx)
         !isTesSuccess(err))
         return err;
 
+    if (ctx.tx.isFieldPresent(sfDomainID))
+    {
+        if (!permissioned_dex::accountInDomain(
```

Comment:
IMO, it'd be better to have a `Domain` type, and then you can have instance methods like `accountInDomain` and `offerInDomain` just to make everything is strong typed rather than having a uint256_t here